src/Ethereum/Signer.cpp

```diff
@@ -85,31 +83,34 @@ Data addressStringToData(const std::string& asString) {
     return asData;
 }
 
-Transaction Signer::build(const Proto::SigningInput &input) {
+std::shared_ptr<TransactionBase> Signer::build(const Proto::SigningInput& input) {
+    return buildLegacy(input);
```

Comment:
can we avoid using `legacy` or `new`? because the meaning will be changed again, `build` or `buildForEIP2718`

Reply:
Valid point, but naming is difficult...  I used nomenclature from the EIP.  I'm considering: 
PreEIP2718--EIP2718, Original--Enveloped, NonEnveloped--Enveloped, NonTyped--Typed

Reply:
Enveloped / NonEnveloped looks good to me

Reply:
I used NonTyped vs. Typed, having an explicit transaction type is the heart of EIP2718